frontend/src/app/sys-check/welcome/welcome.component.ts

```diff
@@ -42,6 +42,7 @@ export class WelcomeComponent implements OnInit {
       this.getNavigatorInfo();
       this.getBrowserPluginInfo();
       this.getBrowserRating();
+      this.ds.questionnaireReport.length = 0;
```

Comment:
Auf die Art einen Array zurückzusetzen ist zwar technisch möglich (wie ich gerade gelesen habe), scheint mir aber keine so gute Idee. Ich würde generell abraten das length-Attribut manuell zu manipulieren. Das sollte m.E. auch read-only sein. Wie dem auch sei...zum Array leer machen würde ich immer "= []" nehmen.

Zusätzlich stellt sich hier die Frage, ob es guter Stil ist, dass man von außen eine Variable in einer anderen Klasse manipuliert und nicht dafür Methoden baut, aber darüber lässt sich streiten. Ich hätte das an der Stelle aber gleich mitverbessert.

Reply:
Der Unterschied ist, dass mit = [] eine neue Objekt-Instanz erzeugt wird, wie length=0 die gleiche beibehalten wird. Spielt hier absolut keine Role, in bestimmten Kontexten aber schon, weshalb mMn beide Methoden ihre Berechtigung haben.

Reply:
Ja das fiel mir dann auch noch ein, dass so die Instanz erhalten bleibt. Nur fällt mir kein sinnvoller Fall ein, wo man das brauchen könnte. Meistens will man eh nur pushen oder popen. Der Fall, dass man x Elemente am Ende des Array wegschneiden will, ist mir bisher nicht untergekommen. Wenn man mein das tun zu müssen, dann sollte man lieber drüber nachdenken, ob ein Array der passende Datentyp ist.